src/mc.rs

```diff
@@ -2369,6 +2369,37 @@ impl Rav1dMCDSPContext {
             });
         }
 
+        #[cfg(all(target_arch = "aarch64", feature = "asm_arm64_sve2"))]
+        if BD::BITDEPTH == 16 {
+            if !flags.contains(CpuFlags::SVE2) {
+                return self;
+            }
```

Comment:
```suggestion
        if BD::BITDEPTH == 16 && flags.contains(CpuFlags::SVE2) {
```

Reply:
Can you do the same changes to `DOTPROD` and `I8MM` so that it matches the `mc.h` logic so we stay synced as similar as possible?

Reply:
It looks like the C is changed to me: 

```c
    if (flags & DAV1D_ARM_CPU_FLAG_DOTPROD) {
        init_8tap_fns(neon_dotprod);
    }
```

Reply:
Indeed. Must have been looking at the wrong version of the C file.